modules/gtk/hm.nix

```diff
@@ -29,25 +29,66 @@ in {
         window.background { border-radius: 0; }
       '';
     };
+
+    flatpakSupport.enable = lib.mkEnableOption "support for theming Flatpak apps";
   };
 
-  config = lib.mkIf cfg.enable {
-    # programs.dconf.enable = true; required in system config
-    gtk = {
-      enable = true;
-      font = {
-        inherit (config.stylix.fonts.sansSerif) package name;
-        size = config.stylix.fonts.sizes.applications;
+  config = lib.mkMerge [
+    (lib.mkIf cfg.enable {
+      # programs.dconf.enable = true; required in system config
+      gtk = {
+        enable = true;
+        font = {
+          inherit (config.stylix.fonts.sansSerif) package name;
+          size = config.stylix.fonts.sizes.applications;
+        };
+        theme = {
+          package = pkgs.adw-gtk3;
+          name = "adw-gtk3";
+        };
       };
-      theme = {
-        package = pkgs.adw-gtk3;
-        name = "adw-gtk3";
+
+      xdg.configFile = {
+        "gtk-3.0/gtk.css".source = finalCss;
+        "gtk-4.0/gtk.css".source = finalCss;
       };
-    };
+    })
 
-    xdg.configFile = {
-      "gtk-3.0/gtk.css".source = finalCss;
-      "gtk-4.0/gtk.css".source = finalCss;
-    };
-  };
+    (lib.mkIf (cfg.enable && cfg.flatpakSupport.enable) ({
+      # Flatpak apps apparently don't consume the CSS config. This workaround appends it to the theme directly.
+      home.file.".themes/${config.gtk.theme.name}".source = pkgs.stdenv.mkDerivation {
+        name = "flattenedGtkTheme";
+        src = "${config.gtk.theme.package}/share/themes/${config.gtk.theme.name}";
+
+        installPhase = ''
+          mkdir $out
+          cp -rL . $out
+
+          config="${config.xdg.configFile."gtk-3.0/gtk.css".source}"
+          cat "$config" >> $out/gtk-3.0/gtk.css
+          cat "$config" >> $out/gtk-4.0/gtk.css
+        '';
+      };
+    } // (let
+        filesystem = "${config.home.homeDirectory}/.themes/${config.gtk.theme.name}:ro";
```

Comment:
Consider handling special cases where `config.home.homeDirectory` or `config.gtk.theme.name` contain special characters, like ` ` (space), `'` (single quote), or `"` (double quotes), by leveraging `lib.escapeShellArg` or quoting the values when necessary.

Reply:
I tried the Flatpak CLI with the path `~/test "'.txt` and it seems like they are not escaped. I'll leave it as is.
```zsh
[~]$ flatpak override -u --filesystem "/home/bricked/test \"'.txt"

[~]$ cat ~/.local/share/flatpak/overrides/global
[Context]
filesystems=/home/bricked/.themes/adw-gtk3:ro;/home/bricked/test "'.txt;

[Environment]
GTK_THEME=adw-gtk3

[~]$ flatpak run --command=bash org.gnome.Calculator 

[📦 org.gnome.Calculator ~]$ ls
'test "'\''.txt'
```

Reply:
> ```nix
> filesystems=/home/bricked/.themes/adw-gtk3:ro;/home/bricked/test "'.txt;
> ```

Seems like `filesystems` is `;`-separated. Could we paranoically escape them? Maybe the flatpak documentation lists the reserved characters we have to escape.

Reply:
Sadly I couldn't find any (obvious) documentation on the file format itself, but I will try the CLI on a filename containing a semicolon and see what it does.

Reply:
Nevermind, there *is* documentation on the file format: https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpakref
But it doesn't really explain the syntax, other than saying that it's using the .ini file format. But that can't be correct, since the .ini file format doesn't support lists and parses semicolons as comments.

Reply:
> Nevermind, there _is_ documentation on the file format: https://docs.flatpak.org/en/latest/flatpak-command-reference.html#flatpakref But it doesn't really explain the syntax, other than saying that it's using the .ini file format. But that can't be correct, since the .ini file format doesn't support lists and parses semicolons as comments.

Maybe we should just ignorantly not escape anything. After all, it would be weird for anyone to attack themselves by calling their home directory `/home/user; echo $GH_TOKEN`.

Reply:
OK I tried the flatpak CLI and it seems to escape some tokens using backslashes:
- `;` becomes `\;`
- single `\` is deleted
- `\\` becomes `\\\\\`
- `; ` becomes `\;\s` (??)
- spaces, brackets and shell variables are not escaped

I don't think it would make sense to implement these edge cases, as there isn't sufficient documentation, so custom implementations might do more harm than good and there isn't a realistic attack surface.

Reply:
Agreed.

Reply:
I just dealt with desktop entries and noticed that they seem to follow the same file format, just like all other freedesktop standards. List formatting is described in https://specifications.freedesktop.org/desktop-entry-spec/latest/value-types.html and seems to validate my observations.